giskard/ml_worker/websocket/listener.py

```diff
@@ -84,70 +182,32 @@ def dispatch_action(callback, ml_worker, action, req):
-    except Exception as e:
-        info: websocket.WorkerReply = websocket.ErrorReply(
-            error_str=str(e), error_type=type(e).__name__, detail=traceback.format_exc()
-        )
-        logger.warning(e)
-    finally:
-        analytics.track(
-            "mlworker:websocket:action",
-            {
-                "name": action.name,
-                "worker": ml_worker.ml_worker_id,
-                "language": "PYTHON",
-                "type": "ERROR" if isinstance(info, websocket.ErrorReply) else "SUCCESS",
-                "action_time": time.process_time() - start,
-                "error": info.error_str if isinstance(info, websocket.ErrorReply) else "",
-                "error_type": info.error_type if isinstance(info, websocket.ErrorReply) else "",
-            },
-        )
 
-    if rep_id:
-        # Reply if there is an ID
-        logger.debug(
-            f"[WRAPPED_CALLBACK] replying {len(info.json(by_alias=True))} {info.json(by_alias=True)} for {action.name}"
-        )
-        # Message fragmentation
-        FRAG_LEN = max(ml_worker.ws_max_reply_payload_size, MAX_STOMP_ML_WORKER_REPLY_SIZE)
-        payload = info.json(by_alias=True) if info else "{}"
-        frag_count = math.ceil(len(payload) / FRAG_LEN)
-        for frag_i in range(frag_count):
-            ml_worker.ws_conn.send(
-                f"/app/ml-worker/{ml_worker.ml_worker_id}/rep",
-                json.dumps(
-                    {
-                        "id": rep_id,
-                        "action": action.name,
-                        "payload": fragment_message(payload, frag_i, FRAG_LEN),
-                        "f_index": frag_i,
-                        "f_count": frag_count,
-                    }
-                ),
-            )
-
-        analytics.track(
-            "mlworker:websocket:action:reply",
-            {
-                "name": action.name,
-                "worker": ml_worker.ml_worker_id,
-                "language": "PYTHON",
-                "action_time": time.process_time() - start,
-                "is_error": isinstance(info, websocket.ErrorReply),
-                "frag_len": FRAG_LEN,
-                "frag_count": frag_count,
-                "reply_len": len(payload),
-            },
-        )
+    logger.debug("Submitting for action %s '%s' into the pool with %s", action.name, callback.__name__, params)
+    future = call_in_pool(
+        parse_and_execute,
+        callback=callback,
+        action=action,
+        params=params,
+        ml_worker=MLWorkerInfo(ml_worker),
+        client_params=client_params,
+    )
+    future.add_done_callback(wrapped_handle_result(action, ml_worker, start, rep_id))
+    log_pool_stats()
 
-    # Post-processing of stopWorker
-    if action == MLWorkerAction.stopWorker and ml_worker.ws_stopping is True:
-        ml_worker.ws_conn.disconnect()
+    return
 
 
 def websocket_actor(action: MLWorkerAction):
```

Comment:
Did you decide not to add a `execute_in_pool` boolean flag to this decorator?

Reply:
Good idea for some quick and short task

Reply:
Yeah, I forgot about that since I had issues with the enron. Just added